core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -1268,6 +1268,19 @@ object TestUtils extends Logging {
     assertTrue(s"$message failed with exception(s) $exceptions", exceptions.isEmpty)
 
   }
+
+  def consumeMessages(servers: Seq[KafkaServer], topic: String, numMessages: Int): List[ConsumerRecord[Array[Byte], Array[Byte]]] = {
+    val records = new ArrayList[ConsumerRecord[Array[Byte], Array[Byte]]]()
+    val consumer = createNewConsumer(
+            TestUtils.getBrokerListStrFromServers(servers),
+            securityProtocol = SecurityProtocol.PLAINTEXT)
+    consumer.subscribe(Collections.singleton(topic))
+    do {
+      for (record <- consumer.poll(50).asScala)
+        records.add(record)
+    } while (records.size < numMessages)
+    records.asScala.toList
```

Comment:
I refactored this into 2 separate methods, made sure the consumer was closed, checked the exact number of messages and added a timeout.